src/source.cpp

```diff
@@ -44,9 +45,9 @@ std::vector<SourceDistribution> external_sources;
 
 namespace {
 
-using sample_t = Particle::Bank (*)(uint64_t* seed);
-sample_t custom_source_function;
 void* custom_source_library;
+std::string custom_source_parameters = "";
```

Comment:
Strings are automatically initialized to "", so no need to put that explicitly
```suggestion
std::string custom_source_parameters;
```